### PR TITLE
Fix: content and aesthetics globally to add help text and action icons 

### PR DESCRIPTION
**Pull request by SirineMhedhbi**
## Context
 See: https://docs.google.com/spreadsheets/d/1lEdrxJKKOJYbSYH_Ptzu2dUliHAKcFTirrw8vs9Hcuk/edit#gid=713725591
## Changes

- [x] Missing (i) for blocks: relations and submissions. 

- [x] Add "(i)" for blocks: relations and submissions. Include: "Relation with other ontologies either in PORTALNAME (and set portal name color to match the portal) or outside (and use the blue color used in the graphic)." 
image

- [x] All external link icons must open in a new browser tab. 
image

- [x] Add a code icon </> next to "Submissions" and "Metrics" that directs to the corresponding web service calls in a new tab. 
image

- [x] Add the </> icon to the right of the "Download" button for each line in the submissions block and ensure that this button opens the JSON corresponding to the metadata record of that submission.
 - [x] Capitalize titles in support. 
 image

- [x] Fix the display of portal_name (Register).
image

- [x] In concept details, localize dates 
image

- [x] remove : Asterix after 'feedback' leads nowhere.
image

- [x] Add fullscreen view of the ontology relation chart
image
image

- [x]  Change the 'Changes notes' block to span multiple lines when filled with a large amount of text
image

- [x]  Center : Admin page

- [x]  Tab 'Property' on the summary page => ID should come first at the top, not last. Also, 'Labels' should be 'Preferred name'
image

- [x]  Change the sentence at the top (which remains on all tabs)
- [x]  Add text under license : "PORTALNAME requires an URI for the license. If you do not find your chice here, please pickup an URI from https://rdflicense.linkeddata.es/"
image

- [x]  Add under deprecated: "An ontology with status retired shall necessarily be also deprecated, but not the oposite.” + Minimize the tooltip info as others

image

- [x]  Change : "Sort by class count" to "Sort by classes count”
image

- [x]  Total score is a butoon that click does not bring anywhere.( no cursor)
image

- [x]  change "See details" de O'FAIRe => "O'FAIRe FAIRness assessment questions”
image

- [x]  change "Décconnexion" --> Déconnexion
image

- [x]  Change Actions (submissions) and delete downloads
   - user is not logged: 
   
image

   - user is logged: 
   
image

- [x]  add under Known usage : "Consider also declaring the projects that are using the ontology.”
image

- [x]  add "Bug database" : "Some ontology feedback and notes features are only possible if a GitHub repository is informed.”
image

- [x]  add under Creator : "The following properties take for value an "agent" in PORTALNAME (either a person or an organization). These agents are shared over all the ontologies and suggested with autocompletion if they already exist. Editing an agent here will change it to all the ontologies that agent is involved in.”
image

- [x] Could be nice if https://stageportal.lirmm.fr/ontologies/AGRO#projects_section would actually already open the Project drop down section on the Summary page (so it could be sue a s link from Browse page) 

image

### CODE REVIEW

File: app/helpers/submission_inputs_helper.rb [hunk 1]
```diff
@@ -399,23 +399,22 @@ def generate_ontology_select_input(name, label, selected, multiple)
...
 
-  def generate_list_text_input(attr)
+  def generate_list_text_input(attr, helper_text: nil)
     label = attr_header_label(attr)
     values = attr.values || ['']
     name = attr.name
-    generate_list_field_input(attr, name, label, values) do |value, row_name, id|
-      text_input(label: '', name: row_name, value: value)
+    generate_list_field_input(attr, name, label, values, helper_text: helper_text) do |value, row_name, id|
+      text_area_tag(row_name, value, class: 'input-field-component', label: '')
```
Comment (syphax-bouazzouni) [inline app/helpers/submission_inputs_helper.rb:407]: why this change ?

File: app/views/admin/index.html.haml [hunk 2]
```diff
@@ -38,8 +38,6 @@
           %table#adminOntologies.zebra{:cellpadding => "0", :cellspacing => "0", :width => "100%"}
       - t.item_content do
 
-        %div#renew-license-notice
```
Comment (syphax-bouazzouni) [inline app/views/admin/index.html.haml:41]: don't remove this

File: app/views/ontologies/sections/metadata/_ontology_relations_network.html.haml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 - unless @ontology_relations_data.nil? || @ontology_relations_data.empty?
-  = render SummarySectionComponent.new(title: 'Ontology relations network') do
+  = render SummarySectionComponent.new(title: 'Ontology relations network', link: "", link_title: "Relation with other ontologies either in <span style='color: #3CB371 !important;'>#{portal_name}</span> or <span style='color:#007bff;'>outside</span>") do
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/metadata/_ontology_relations_network.html.haml:2]: create a helper for the link_title, instead of writing here

File: app/views/submissions/_submissions.html.haml [hunk 4]
```diff
@@ -21,45 +21,45 @@
...
+                  .dropdown-menu
+                    - links = download_link(sub,@ontology)
+                    - links.each do |value|
+                      - link,label = value.values               
+                      %a.dropdown-item{ href: link }= label
+                 
+                -# = raw download_link(sub, @ontology)
+                - if @ontology.admin?(session[:user])
+                  %div.d-flex
+                    = link_to("#{@ontology.id}/submissions/#{sub.submissionId}?display=all", target: "_blank",title: 'Go to API') do
```
Comment (syphax-bouazzouni) [inline app/views/submissions/_submissions.html.haml:54]: create a helper for each of these.
And they need to be re-usable, we will use them everywhere in the future, as, edit, remove button

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

File: app/helpers/submission_inputs_helper.rb (hunk 1 above)
Comment (SirineMhedhbi) [reply app/helpers/submission_inputs_helper.rb:407]: It is to enlarge the 'change notes' section in order to write a long text
image

File: app/views/ontologies/sections/metadata/_ontology_relations_network.html.haml (hunk 3 above)
Comment (SirineMhedhbi) [reply app/views/ontologies/sections/metadata/_ontology_relations_network.html.haml]: done

File: app/assets/stylesheets/components/tabs_container.scss [hunk 5]
```diff
@@ -25,7 +25,7 @@
   font-size: 16px;
   font-weight: 400;
   color: #5e5e5e;
-  margin-right: 50px;
+  margin-right: 25px;
```
[the remark this answers is not in the document]
Comment (SirineMhedhbi) [reply app/assets/stylesheets/components/tabs_container.scss:28]: done

File: app/assets/stylesheets/components/tabs_container.scss [hunk 6]
```diff
@@ -46,7 +46,7 @@
 }
 
 .tabs-container .tab-items .active {
-  font-weight: 500;
```
Comment (syphax-bouazzouni) [inline app/assets/stylesheets/components/tabs_container.scss:49]: don't change this

File: app/components/concept_details_component/concept_details_component.html.haml [hunk 7]
```diff
@@ -6,6 +6,7 @@
     - if @bottom_keys.present?
       = render DropdownContainerComponent.new(title: 'Raw data', id: "accordion-#{@id}") do
         - top_set, leftover_set, bottom_set = filter_properties(@top_keys, @bottom_keys, @exclude_keys, prefix_properties(@concept_properties))
+        - leftover_set = convert_dates(leftover_set)
```
Comment (syphax-bouazzouni) [inline app/components/concept_details_component/concept_details_component.html.haml:9]: if you use `convert_dates` only here, don't put the function in the `application_helper.rb`, just put the function `concept_details_component.rb`

File: app/helpers/metadata_helper.rb [hunk 8]
```diff
@@ -89,4 +89,13 @@ def display_attribute(metadata, value)
     end
   end
 
+  def generate_link_title
```
Comment (syphax-bouazzouni) [inline app/helpers/metadata_helper.rb:92]: put it in `ontologies_helper.rb`

File: app/helpers/application_helper.rb [hunk 9]
```diff
@@ -573,6 +573,16 @@ def prefix_properties(concept_properties)
     modified_properties
   end
 
+  def convert_dates(hash)
```
Comment (syphax-bouazzouni) [inline app/helpers/application_helper.rb:576]: put it in the `concept_details_component.rb`

File: app/components/summary_section_component.rb [hunk 10]
```diff
@@ -3,15 +3,28 @@
...
   renders_many :action_links
 
-  def initialize(title: , link: nil, link_title: nil, show_card:  true)
+  def initialize(title: , link: nil, link_title: nil, show_card:  true, show_icon: false, service_link: nil)
     super
     @title = title
     @link = link
     @link_title = link_title
     @show_card = show_card
+    @show_icon = show_icon
```
Comment (syphax-bouazzouni) [inline app/components/summary_section_component.rb:12]: here you can remove `show_icon` and jus use `service_link`, as `@show_icon = !service_link.nil?` , no ?

File: app/helpers/submission_inputs_helper.rb (hunk 1 above)
Comment (syphax-bouazzouni) [reply app/helpers/submission_inputs_helper.rb:407]: yes, but this will make all the `text_input` with a big `text area`.
This change needs to affect only changes notes

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

File: app/components/summary_section_component.rb (hunk 10 above)
Comment (SirineMhedhbi) [reply app/components/summary_section_component.rb:12]: I didn't understand what you meant exactly

Comment (syphax-bouazzouni) [reply app/components/summary_section_component.rb:12]: remove the argument `show_icon`, you don't need it, because `service_link` is enaugh

File: app/assets/stylesheets/components/tabs_container.scss (hunk 6 above)
Comment (SirineMhedhbi) [reply app/assets/stylesheets/components/tabs_container.scss]: done

File: app/components/concept_details_component/concept_details_component.html.haml (hunk 7 above)
Comment (SirineMhedhbi) [reply app/components/concept_details_component/concept_details_component.html.haml:9]: done

File: app/helpers/metadata_helper.rb (hunk 8 above)
Comment (SirineMhedhbi) [reply app/helpers/metadata_helper.rb]: done

File: app/helpers/application_helper.rb (hunk 9 above)
Comment (SirineMhedhbi) [reply app/helpers/application_helper.rb]: done

File: app/views/fair_score/_fair_service_header.html.haml [hunk 11]
```diff
@@ -3,4 +3,4 @@
   = beta_badge
 %span
   = render Display::InfoTooltipComponent.new(text: t("view_fair_scores_definitions"))
-  = fairness_link
```
Comment (syphax-bouazzouni) [inline app/views/fair_score/_fair_service_header.html.haml]: here update the fairness_link help to put inside your component

File: app/views/home/index.html.haml [hunk 12]
```diff
@@ -84,7 +84,7 @@
         %div.align-items
           %h4.margin-items= t('home.fairness')
           = render Display::InfoTooltipComponent.new(text: "You are seing the average scores for all the public ontologies in AgroPortal. FAIR scores are computed with the O'FAIRe methodology. More details here: https://github.com/agroportal/fairness")
-          = fairness_link
```
Comment (syphax-bouazzouni) [inline app/views/home/index.html.haml]: here update the fairness_link help to put inside your component

File: app/views/submissions/_submissions.html.haml [hunk 13]
```diff
@@ -21,45 +21,43 @@
...
+                    - links = download_link(sub,@ontology)
+                    - links.each do |value|
+                      - link,label = value.values               
+                      %a.dropdown-item{ href: link }= label
+                 
+                -# = raw download_link(sub, @ontology)
+                - if @ontology.admin?(session[:user])
+                  %div.d-flex
+                    %div{style: 'margin-right: 10px;'}
+                      = render IconWithTooltipComponent.new(icon: "json.svg",link: "#{@ontology.id}/submissions/#{sub.submissionId}?display=all", target: '_blank', title: 'Go to API')
```
Comment (syphax-bouazzouni) [inline app/views/submissions/_submissions.html.haml:55]: still, add help to wrap each of the actions 
```ruby
edit_button(link:, title: )
delete_button(link:, title: )
service_button(link:, title: )
```

File: app/components/layout/reveal_component/reveal_component_controller.js [hunk 14]
```diff
@@ -18,6 +31,15 @@ export default class extends Controller{
         }
     }
 
+    select(event){
+        console.log('woqsfqsdfqsdfrking')
+        let myItems = document.getElementsByClassName(event.target.parentNode.parentNode.getAttribute('data-items'))
```
Comment (syphax-bouazzouni) [inline app/components/layout/reveal_component/reveal_component_controller.js:36]: don't use `myItems` like this, use the stimulus targets (`item` in this case);

File: app/components/layout/reveal_component/reveal_component_controller.js [hunk 15]
```diff
@@ -8,6 +8,19 @@ export default class extends Controller{
...
+    connect(){
+        let element
+        if (
+            this.element.childNodes[0] &&
+            this.element.childNodes[0].childNodes[0] &&
+            this.element.childNodes[0].childNodes[0].childNodes[3]
+          ){
+            element = this.element.childNodes[0].childNodes[0].childNodes[3]
+            if(element.tagName == "SELECT"){
+                document.getElementById(element.value).classList.remove(this.hiddenClassValue)
```
Comment (syphax-bouazzouni) [inline app/components/layout/reveal_component/reveal_component_controller.js:20]: and in a stimulus controller, don't use `document.getElementById` and `document.getElementsByClassName`;
instead, use this.element and stimulus targets

File: app/components/summary_section_component/summary_section_component.html.haml [hunk 16]
```diff
@@ -5,7 +5,8 @@
     %span
       = link_to(@link, target: "_blank") do
         = render Display::InfoTooltipComponent.new(text: @link_title)
-
+  - if show_icon?
+    = render IconWithTooltipComponent.new(icon: "json.svg",link: @service_link, target: '_blank', title: 'Go to API', size:'small', style: 'line-height: 0; margin-top: 3px;margin-left: 6px;')
```
Comment (syphax-bouazzouni) [inline app/components/summary_section_component/summary_section_component.html.haml:9]: wrap this into a helper for readability

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

**Review by syphax-bouazzouni — APPROVED** [review]

**Comment by jonquet** [comment]
Metrics specific popup windows does not show anything. Tested with https://stageportal.lirmm.fr/ontologies/FCU

Capture d’écran 2024-01-26 à 00 01 06
=> missing "i" in including

Capture d’écran 2024-01-26 à 00 01 55
=> portal name is not genralized

Capture d’écran 2024-01-26 à 00 04 26
=> subject can be removed from "Project and usage information" block as its now available in "Categories and subjects" in the General information block.

Capture d’écran 2024-01-26 à 00 09 54
=> missing a p in opposite (oups!) 

Capture d’écran 2024-01-26 à 00 11 22
=> no capital "P" at please
